src/btree/bt_cursor.c

```diff
@@ -2114,3 +2114,71 @@ __wt_btcur_close(WT_CURSOR_BTREE *cbt, bool lowlevel)
 
     return (ret);
 }
+
+/*
+ * __wt_btcur_bounds_row_position --
+ *     A unpositioned bounded cursor need to start its cursor next and prev walk from the lower or
+ *     upper bound depending on which direction it is going. This function calls cursor row search
+ *     to position the cursor appropriately.
+ */
+int
+__wt_btcur_bounds_row_position(
+  WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, bool next, bool *need_walk)
```

Comment:
There is a protocol for return parameters, to add a `p` suffix (standing for pointer). Could you rename `need_walk` to `need_walkp`?